### PR TITLE
improvement: validates attributes and relationships have unique names

### PR DESCRIPTION
**Pull request by tlietz**
Issue: https://github.com/ash-project/ash/issues/214
# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [x] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

**Comment by zachdaniel** [comment]
This looks excellent! The only thing I'd say is that we should include calculations and aggregates as well, as they should all be unique.

**Comment by tlietz** [comment]
@zachdaniel Sounds good, should those all be in the same transformer file?

**Comment by zachdaniel** [comment]
Yeah, what you'd want to do is group up all of them with the same name, and then say something simple like: "There are multiple fields(attributes, calculations, aggregates, and relationships) that share the name <foo>".

**Comment by tlietz** [comment]
Since this test case is including all fields that are a part of a resource, should it be moved out of `test/resource/relationships/belongs_to_test.exs` and into something like `test/resource/validations_test.exs` instead?

**Comment by zachdaniel** [comment]
Yeah, good idea! 👍🏻

**Comment by zachdaniel** [comment]
Actually, `validations_test` is for actual validations. I'd make a new file `resource_test.exs`.

**Comment by tlietz** [comment]
Alright, this should be ready for review now.

File: lib/ash/resource/dsl.ex [hunk 1]
```diff
@@ -1110,7 +1110,8 @@ defmodule Ash.Resource.Dsl do
     Ash.Resource.Transformers.DefaultPrimaryKey,
     Ash.Resource.Transformers.DefaultAccept,
     Ash.Resource.Transformers.SetTypes,
-    Ash.Resource.Transformers.ValidateRelationshipAttributes
+    Ash.Resource.Transformers.ValidateRelationshipAttributes,
```
Comment (zachdaniel) [inline lib/ash/resource/dsl.ex:1113]: we can remove `ValidateRelationshipAttributes` now right?

**Comment by zachdaniel** [comment]
Had one comment, otherwise this is good to go. Great work!

File: lib/ash/resource/dsl.ex (hunk 1 above)
Comment (tlietz) [reply lib/ash/resource/dsl.ex]: Yes we can, good catch!

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**